inception/inception-imls-external/src/main/java/de/tudarmstadt/ukp/inception/recommendation/imls/external/ExternalRecommender.java

```diff
@@ -129,26 +128,26 @@ public void train(RecommenderContext aContext, List<CAS> aCasses) throws Recomme
 
         trainingRequest.setDocuments(documents);
 
-        HttpUrl url = HttpUrl.parse(traits.getRemoteUrl()).newBuilder().addPathSegment("train")
-                .build();
-        RequestBody body = RequestBody.create(JSON, toJson(trainingRequest));
-        Request request = new Request.Builder().url(url).post(body).build();
-
-        try (Response response = sendRequest(request)) {
-            if (response.code() == HTTP_TOO_MANY_REQUESTS) {
-                LOG.info("External recommender is already training");
-            }
-            // If the response indicates that the request was not successful,
-            // then it does not make sense to go on and try to decode the XMI
-            else if (!response.isSuccessful()) {
-                int code = response.code();
-                String responseBody = getResponseBody(response);
-                String msg = format("Request was not successful: [%d] - [%s]", code, responseBody);
-                throw new RecommendationException(msg);
-            }
-
-            aContext.put(KEY_TRAINING_COMPLETE, true);
+        HttpRequest request = HttpRequest.newBuilder() //
+                .uri(URI.create(traits.getRemoteUrl()).resolve("train")) //
```

Comment:
Uri::resolve is dangerous I think if the last char is not a slash:

```
import java.net.URI;
public class Main{

     public static void main(String []args){
        System.out.println(URI.create("http://example.com/foo/bar/baz").resolve("train"));
     }
}
```

`> http://example.com/foo/bar/train`

Reply:
Looks like `URI.create()` normalizes paths and there is also a handy `appendIfMissing` method in `StringUtils` - so I'll use that to ensure the URL ends in a slash before resolving.